### PR TITLE
feat: add IPv6 support

### PR DESCRIPTION
**Pull request by italodeverdade**
Hey, first of all thanks for the amazing project! :) 

I'm implementing a python project using `simple-websocket` on Railway and realized that `simple-websocket` don't support IPv6 yet. So i took the time to change the `AF_INET` to `AF_INET6`. Following the documentation specifics about the differences between both it should enough to use just `AF_INET6` for now :)

> AF_INET6 sockets provide support for Internet Protocol version 6 (IPv6) 128 bit (16 byte) address structures. Programmers can write applications using the AF_INET6 address family to accept client requests from either IPv4 or IPv6 nodes, or from IPv6 nodes only.
> 
> Like AF_INET sockets, AF_INET6 sockets can be either connection-oriented (type SOCK_STREAM) or connectionless (type SOCK_DGRAM). Connection-oriented AF_INET6 sockets use TCP as the transport protocol. Connectionless AF_INET6 sockets use User Datagram Protocol (UDP) as the transport protocol. When you create an AF_INET6 domain socket, you specify AF_INET6 for the address family in the socket program. AF_INET6 sockets can also use a type of SOCK_RAW. If this type is set, the application connects directly to the IP layer and does not use either the TCP or UDP transport.

I could be wrong here, but prefer to open the PR here so we can may add support in future? Thanks!

### CODE REVIEW

File: src/simple_websocket/ws.py [hunk 1]
```diff
@@ -397,7 +397,7 @@ def __init__(self, url, subprotocols=None, headers=None,
         elif isinstance(headers, list):
             self.extra_headeers = headers
 
-        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```
Comment (miguelgrinberg) [inline src/simple_websocket/ws.py:400]: Will this allow the client to connect to a IPv4 only server? I'm not really sure this is true for every platform, especially on Windows. I would prefer an option passed in the constructor to select IPv6.

Comment (italodeverdade) [reply src/simple_websocket/ws.py:400]: @miguelgrinberg i tested on MacOS and on Linux and both work for IPV4 and IPV6 but didn't tried on Windows, indeed. I just changed my last code to use `address_family` parameter in the class constructor and use the default as `socket.AF_INET` so no break changes in the behavior.

Thanks for the feedback!

**Comment by miguelgrinberg** [comment]
Couple of notes.

First, the async client would need to be changed to support the same functionality, and ideally with the same change in the constructor arguments.

Second, I'm thinking there is another possible way to handle this, which would involve calling `getaddrinfo(host, port, type=socket.SOCK_STREAM)` and letting the system decide itself between IPv4 and IPv6. The `address_family` would only be needed in case you want to force one type or the other. The [getaddrinfo](https://docs.python.org/3/library/socket.html#socket.getaddrinfo) function returns the arguments to pass to the socket constructor and the connect call. WDYT?

**Comment by italodeverdade** [comment]
Hey @miguelgrinberg, first sorry to take longer to work on your feedback. Work has been crazy these days 🥲 

So, i have checked the async client but it seems it uses different implementation without the need to pass address_family there so i assume that the async client handles it automatically? I made the changes but to be honest in a moment it feels wrong so let me know if you really need it there :) 

I fixed also a typo on `extra_headers` parameters.

**Comment by miguelgrinberg** [comment]
I have implemented a simpler solution based on `getaddrinfo()` as I described above. The choice between ipv4 or ipv6 is made by this call.

**Pull request closed without merging**